fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/IndexingByIndex.java

```diff
@@ -192,7 +191,7 @@ private CompletableFuture<Boolean> buildRangeOnly(@Nonnull FDBRecordStore store,
                                           lastResult.get().get().getIndexEntry().getKey() :
                                           rangeEnd)
                     .thenCompose(cont -> rangeSet.insertRange(store.ensureContextActive(), packOrNull(rangeStart), packOrNull(cont), true)
-                                .thenApply(ignore -> !Objects.equals(cont, rangeEnd)));
+                                .thenApply(ignore -> !allRangesExhausted(cont, rangeEnd)));
```

Comment:
This condition is one of the places where using `allRangesExhausted` looks wrong to me (both now and before this PR). Note that `rangeEnd` was computed from reading the `rangeSet` above, so this lambda will now never return `false` if there's every any range in the range set. Before this PR, it was also wrong, but it would return `false` as soon as it completes the first range. But I think it's working, in practice, because `IndexingByIndex` builds would always be structured in a way that they'd have a `null` value as the end of at least one range. The way it reads the range set and only appears to grab the first element is....also kind of suspicious, though that's not affected by this change as directly